`dockerfile verify` command fails when stage names are used in `FROM`

**Description**

When a Dockerfile contains a `FROM` statement using a stage name instead of image, `dockerfile verify` fails because `cosign` tries to pull the stage name as if it was an image.

For example, using this Dockerfile:
```dockerfile
FROM cgr.dev/chainguard/static:latest AS base_image
# some customization here
FROM base_image
ENTRYPOINT "/executable"
```

running `cosign dockerfile verify Dockerfile --certificate-oidc-issuer=https://token.actions.githubusercontent.com --certificate-identity=https://github.com/chainguard-images/images/.github/workflows/release.yaml@refs/heads/main` ends with an error:

```
Extracted image(s): cgr.dev/chainguard/static:latest, base_image
(...)
Error: GET https://index.docker.io/v2/library/base_image/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:library/base_image Type:repository]]
main.go:69: error during command execution: GET https://index.docker.io/v2/library/base_image/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:library/base_image Type:repository]]
```

Commenting out `FROM base_image` in `Dockerfile` makes it work.

**Version**
```
GitVersion:    v2.4.0
GitCommit:     b5e7dc123a272080f4af4554054797296271e902
GitTreeState:  clean
BuildDate:     2024-08-06T21:39:53Z
GoVersion:     go1.22.5
Compiler:      gc
Platform:      linux/amd64
```

--- a/cmd/cosign/cli/dockerfile/verify.go
+++ b/cmd/cosign/cli/dockerfile/verify.go
@@ -93,9 +93,12 @@ func (fc *finderCache) getImagesFromDockerfile(ctx context.Context, dockerfile i
 		lineUpper := strings.ToUpper(line)
 		switch {
 		case strings.HasPrefix(lineUpper, "FROM"):
-			switch image := fc.getImageFromLine(line); image {
-			case "scratch":
+			image := fc.getImageFromLine(line)
+			switch {
+			case image == "scratch":
 				ui.Infof(ctx, "- scratch image ignored")
+			case fc.isStage(image):
+				ui.Infof(ctx, "- stage reference ignored: %s", image)
 			default:
 				images = append(images, image)
 			}

--- a/cmd/cosign/cli/dockerfile/verify_test.go
+++ b/cmd/cosign/cli/dockerfile/verify_test.go
@@ -118,6 +118,13 @@ COPY --from=prepare /app /app`,
 			expected: []string{"gcr.io/someorg/coolimage", "gcr.io/someorg/someimage"},
 		},
 		{
+			name: "from-references-previous-stage",
+			fileContents: `FROM gcr.io/someorg/baseimage AS base_image
+			RUN customize
+			FROM base_image`,
+			expected: []string{"gcr.io/someorg/baseimage"},
+		},
+		{
 			name: "gauntlet",
 			fileContents: `FROM gcr.io/${TEST_IMAGE_REPO_PATH}/one AS one
 RUN script1

